pkg/controllers/user/logging/clusterlogging.go

```diff
@@ -124,18 +138,52 @@ func (c *ClusterLoggingSyncer) createOrUpdateClusterConfigMap() error {
 	return utils.UpdateConfigMap(loggingconfig.ClusterConfigPath, loggingconfig.ClusterLoggingName, "cluster", c.configmaps)
 }
 
-func (c *ClusterLoggingSyncer) updateEmbeddedEndpoint() {
-	timeout := 3 * time.Minute
-	syncInterval := 10 * time.Second
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
-	go func(ctx context.Context, cancel context.CancelFunc) {
-		for range ticker.Context(ctx, syncInterval) {
-			err := utils.UpdateEmbeddedEndpoint(c.podLister, c.serviceLister, c.clusterLoggings)
-			if err != nil {
-				logrus.Debugf("updateEmbeddedEndpoint failed, %v", err)
-			} else {
-				cancel()
-			}
+func (c *ClusterLoggingLifecycle) provision(obj *v3.ClusterLogging) (newobj *v3.ClusterLogging, err error) {
+	if v3.LoggingConditionProvisioned.IsTrue(obj) {
```

Comment:
why we need to check LoggingConditionProvisioned here?

provision are called inside v3.LoggingConditionProvisioned.Do already.